### PR TITLE
Add AM dataset `argmicro`

### PR DESCRIPTION
**Pull request by idalr**
As described in #10, we add a new argument mining dataset - ArgMicro.
This includes:

 - [x] PIE dataset builder: pie/argmicro.py
 - [x] dataset card: pie/argmicro/README.md
 - [x] test file: test_argmicro.py
 - [x] new dataset card for HF: [README.md](https://huggingface.co/datasets/DFKI-SLT/argmicro/blob/main/README.md) in [282733d](https://huggingface.co/datasets/DFKI-SLT/argmicro/commit/282733d6d57243f2a202d81143c4e31bb250e663)

### CODE REVIEW

File: dataset_builders/pie/argmicro/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+[ArgMicro Huggingface dataset loading script](https://huggingface.co/datasets/DFKI-SLT/argmicro).
+
+## Data Schema
+
+The document type for this dataset is `ArgMicroDocument` which defines the following data fields:
+
+- `id` (str)
+- `text` (str)
+- `topic_id` (str, optional)
+- `stance` (str, optional)
```
Comment (ArneBinder) [inline dataset_builders/pie/argmicro/README.md:13]: This is now also an annotation layer, see https://github.com/ArneBinder/pie-datasets/pull/56/commits/52afbb66e79d2cfb23084ab98de398a60123a7d6

File: tests/dataset_builders/pie/test_argmicro.py [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+DATASET_NAME = "argmicro"
+SPLIT_SIZES = {"train": 112}
+DATA_PATH = FIXTURES_ROOT / "dataset_builders" / "arg-microtexts-master.zip"
+HF_DATASET_PATH = ArgMicro.BASE_DATASET_PATH
+PIE_DATASET_PATH = PIE_BASE_PATH / DATASET_NAME
+
+
+@pytest.fixture(scope="module", params=["en", "de"])
+def language(request):
+    return request.param
```
Comment (ArneBinder) [inline tests/dataset_builders/pie/test_argmicro.py:38]: rename this fixture to `dataset_variant` and get values directly from the dataset builder, see [here](https://github.com/ArneBinder/pie-datasets/blob/main/tests/dataset_builders/pie/test_tacred.py#L35-L37)

File: dataset_builders/pie/argmicro/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- `metadata` (dictionary, optional)
+
+and the following annotation layers:
+
+- `edus` (annotation type: `Span`, target: `text`)
+- `adus` (annotation type: `LabeledAnnotationCollection`, target: `edus`)
+- `relations` (annotation type: `MultiRelation`, target: `adus`)
```
Comment (ArneBinder) [inline dataset_builders/pie/argmicro/README.md:20]: both of these types are a bit special, can you add a subitem below each of them that shortly explains what they are?

File: dataset_builders/pie/argmicro/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+See [here](https://github.com/ChristophAlt/pytorch-ie/blob/main/src/pytorch_ie/annotations.py) for the annotation type definitions.
+
+## Document Converters
+
+The dataset provides document converters for the following target document types:
+
+- `pytorch_ie.documents.TextDocumentWithLabeledSpansAndBinaryRelations`
```
Comment (ArneBinder) [inline dataset_builders/pie/argmicro/README.md:28]: this is also not trivial how this works, can you add some subitems that shortly explains how we get the spans and binary relations from the given annotations?

File: tests/dataset_builders/pie/test_argmicro.py [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+def generate_document_kwargs(hf_dataset, language):
+    ds = hf_dataset["train"]
+    return ArgMicro(config_name=language)._generate_document_kwargs(ds)
+
+
+def test_hf_dataset(hf_dataset, language, generate_document_kwargs):
+    assert hf_dataset is not None
+    assert {name: len(ds) for name, ds in hf_dataset.items()} == SPLIT_SIZES
+    ds = hf_dataset["train"]
+    # check
```
Comment (ArneBinder) [inline tests/dataset_builders/pie/test_argmicro.py:56]: check what?

File: tests/dataset_builders/pie/test_argmicro.py [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+
+
+@pytest.fixture
+def hf_example(hf_dataset):
+    return hf_dataset["train"][0]
+
+
+def test_hf_example(hf_example, hf_dataset, language, generate_document_kwargs):
+    fixture_path = HF_DS_FIXTURE_DATA_PATH / DATASET_NAME / f"{language}.train.0.json"
+    # hf_example_expected = json.loads(fixture_path.read_text())
```
Comment (ArneBinder) [inline tests/dataset_builders/pie/test_argmicro.py:110]: remove

File: tests/dataset_builders/pie/test_argmicro.py [hunk 7]
```diff
@@ -0,0 +1,549 @@
...
+    hf_example_expected = json.loads(open(fixture_path, encoding="utf-8").read())
```
Comment (ArneBinder) [inline tests/dataset_builders/pie/test_argmicro.py:111]: why not using `json.load`?

File: tests/dataset_builders/pie/test_argmicro.py [hunk 8]
```diff
@@ -0,0 +1,549 @@
...
+        "und",
+        "add",
+        "seg",
+        "seg",
+        "seg",
+        "seg",
+    ]
+
+
+# need to find bug
```
Comment (ArneBinder) [inline tests/dataset_builders/pie/test_argmicro.py:138]: still valid?

**Review by ArneBinder — CHANGES_REQUESTED** [review]
See below.

File: dataset_builders/pie/argmicro/README.md (hunk 1 above)
Comment (idalr) [reply dataset_builders/pie/argmicro/README.md]: edit in 8e97ed3

File: tests/dataset_builders/pie/test_argmicro.py (hunk 2 above)
Comment (idalr) [reply tests/dataset_builders/pie/test_argmicro.py]: edit in 8e97ed3

File: dataset_builders/pie/argmicro/README.md (hunk 3 above)
Comment (idalr) [reply dataset_builders/pie/argmicro/README.md]: edit in 8e97ed3

File: dataset_builders/pie/argmicro/README.md (hunk 4 above)
Comment (idalr) [reply dataset_builders/pie/argmicro/README.md:43]: edit in 8e97ed3

File: tests/dataset_builders/pie/test_argmicro.py (hunk 6 above)
Comment (idalr) [reply tests/dataset_builders/pie/test_argmicro.py]: edit in 8e97ed3

File: tests/dataset_builders/pie/test_argmicro.py (hunk 7 above)
Comment (idalr) [reply tests/dataset_builders/pie/test_argmicro.py]: edit in 8e97ed3

File: tests/dataset_builders/pie/test_argmicro.py (hunk 8 above)
Comment (idalr) [reply tests/dataset_builders/pie/test_argmicro.py]: edit in cced93b

File: tests/dataset_builders/pie/test_argmicro.py (hunk 5 above)
Comment (idalr) [reply tests/dataset_builders/pie/test_argmicro.py]: Not sure when it got there but deleted in cced93b

File: tests/dataset_builders/pie/test_argmicro.py (hunk 2 above)
Comment (idalr) [reply tests/dataset_builders/pie/test_argmicro.py]: edit in cced93b

File: dataset_builders/pie/argmicro/README.md (hunk 3 above)
Comment (ArneBinder) [reply dataset_builders/pie/argmicro/README.md]: hmm this still is not so clear to me. I restructured this a bit and added some TODOs, see https://github.com/ArneBinder/pie-datasets/pull/56/commits/3723f60d2e66a35374fedda810fb10eb5528255b. Please look into them.

File: dataset_builders/pie/argmicro/README.md (hunk 4 above)
Comment (ArneBinder) [reply dataset_builders/pie/argmicro/README.md:44]: As above, see https://github.com/ArneBinder/pie-datasets/pull/56/commits/3723f60d2e66a35374fedda810fb10eb5528255b. Please look into that.

File: dataset_builders/pie/argmicro/README.md (hunk 3 above)
Comment (idalr) [reply dataset_builders/pie/argmicro/README.md]: made further changes in 2a9c40a, please have a look

File: dataset_builders/pie/argmicro/README.md (hunk 4 above)
Comment (idalr) [reply dataset_builders/pie/argmicro/README.md:44]: made further changes in 2a9c40a, please have a look

**Review by ArneBinder — CHANGES_REQUESTED** [review]
Found just one more thing.

File: dataset_builders/pie/argmicro/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+The document type for this dataset is `ArgMicroDocument` which defines the following data fields:
+
+- `id` (str)
+- `text` (str)
+- `topic_id` (str, optional)
+- `metadata` (dictionary, optional)
+
+and the following annotation layers:
+
+- `stance` (annotation type: `Label`, see [here](https://huggingface.co/datasets/DFKI-SLT/argmicro/blob/main/argmicro.py#L35))
```
[the remark this answers is not in the document]
Comment (idalr) [reply dataset_builders/pie/argmicro/README.md]: edit in 8bb3bad

[the remark this answers is not in the document]
Comment (ArneBinder) [reply dataset_builders/pie/argmicro/README.md]: ok, thanks

**Review by ArneBinder — APPROVED** [review]

**Pull request merged**